woocommerce/woocommerce-square
# WooPayments Compatibility

### All Submissions:



---

### Changes proposed in this Pull Request:
PR adds compatibility with the WooPayments extension. it contains the following changes.
- Prevent the express pay buttons from displaying on Square gift card product pages.
- Prevent the express pay buttons from displaying on the cart and checkout page if the card contains a Square gift card product.

Apart from these compatibility changes PR also fixes one issue around split payments to ensure in case of split payment with the gift card only supported payment methods should be displayed.

Closes #121

### Steps to test the changes in this Pull Request:
1. Set up WooPayments extension.
1. Enable Square extension.
1. Create and publish a simple product
        a. Visit the product page: ensure all the express pay buttons appear
        b. Place the item in the cart: visit the cart and ensure all the express pay buttons appear
        c. Place the item in the cart: visit the checkout page and ensure all the express pay buttons appear
1. Create and publish a Gift card product
        a. Visit the product page: ensure no express pay buttons appear from the WooPayments
        b. Place the item in the cart: visit the cart and no express pay buttons appear from the WooPayments
        c. Place the item in the cart: visit the checkout page and no express pay buttons and no payment methods appear from the WooPayments
1. Add a simple product to cart, and make sure it has a price more than $1 (Gift card balance)
1. Go to the checkout page.
1. Add a Gift card on the checkout page and verify that only split payment-supported payment methods are displayed there.

For overall compatibility do the smoke testing on the square extension.

### Changelog entry
> Fix -  Ensure compatibility with WooPayments extension.
